homeassistant/components/mobile_app/notify.py

```diff
@@ -130,9 +136,29 @@ async def async_send_message(self, message="", **kwargs):
             entry = self.hass.data[DOMAIN][DATA_CONFIG_ENTRIES][target]
             entry_data = entry.data
 
-            app_data = entry_data[ATTR_APP_DATA]
-            push_token = app_data[ATTR_PUSH_TOKEN]
-            push_url = app_data[ATTR_PUSH_URL]
+            push_token = None
+            push_url = None
+            platform = None
+            should_encrypt = False
+            cleartext_commands = []
+
+            if entry_data.get(ATTR_PUSH_CONFIG):
+                push_target = entry_data[ATTR_PUSH_CONFIG][0]
+                push_token = push_target[ATTR_PUSH_TOKEN]
+                push_url = push_target[ATTR_PUSH_URL]
+                platform = push_target[ATTR_PLATFORM]
+                should_encrypt = push_target[ATTR_SUPPORTS_ENCRYPTION]
+                cleartext_commands = push_target[ATTR_PUSH_CLEARTEXT_COMMANDS]
+            else:
+                app_data = entry_data[ATTR_APP_DATA]
+                push_token = app_data[ATTR_PUSH_TOKEN]
+                push_url = app_data[ATTR_PUSH_URL]
+
+            if data[ATTR_MESSAGE] in cleartext_commands:
+                data = {"command": data[ATTR_MESSAGE]}
```

Comment:
some of these commands require additional arguments, which may need parsing themselves. for example, `clear_notification`. maybe it would be worth having a 'cleartext_push_url' and 'encrypted_push_url' for this case.